blog/_posts/2019-08-20-multi-container-live-update.md

```diff
@@ -0,0 +1,77 @@
+
+[Here’s an example repo](https://github.com/windmilleng/sidecar_example) you can take for a spin, based on the premise that you might want to iterate on your app and your home-rolled sidecar at the same time. In this Kubernetes YAML, you can see that we have two different containers running on a single pod:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+...
+spec:
+  ...
+    spec:
+      containers:
+        - name: randword
+          image: randword
+          ...
+        - name: log-ingester
+          image: log-ingester
+```
+
+The app is called `randword`, and every second, it logs a random word to a file. The log ingester runs the same pod as the app, reads in the app's logfile by means of a shared volume, and "ingests" the logs (here, just printing them to stdout).
+
+The Tilfile seems simple, because it’s probably what you expected you could write all along, right up until the point you tried, and got the error: "Tilt only supports Live Updates for the first Tilt-built container on a pod." But now it works!
+
+```python
+k8s_yaml('deploy.yaml')
+
+# instructions for how to a. build and b. quickly update the app image
+docker_build('randword', '.', dockerfile='Dockerfile.app',
+             live_update=[
+                 # detect changes to ./app, copy them to the
+                 # running container, and restart the app
+                 sync('./app', '/app'),
+                 run('/restart.sh'),
+             ])
+
+# instructions for how to a. build and b. quickly update the sidecar image
+docker_build('log-ingester', '.', dockerfile='Dockerfile.sidecar',
+             live_update=[
+                 # detect changes to ./app, copy them to the
+                 # running container, and restart the app
+                 sync('./sidecar', '/'),
+                 run('/restart.sh'),
+             ])
+```
+
+Hacking on your sidecar is now just as fast as hacking on your app: make a change in your editor and poof, it’s running in your cluster. Heck, now you can tweak your app and your sidecar at the same and see how they interact! It’s development as it’s meant to be.
+
+## What’s next?
+
+Awareness of multiple containers on a single pod sets us up for awareness of multiple containers across multiple pods--from here, it’ll be much easier to implement Live Updates of a single image across multiple pods. Want to be running multiple replicas of a pod in dev? Let us know!
```

Comment:
I think we should generally link "Let us know"-type statements to https://tilt.dev/contact